s2a_e2e_test.go

```diff
@@ -157,86 +174,145 @@ func runClient(ctx context.Context, t *testing.T, clientS2AAddress, serverAddr s
 	if got, want := resp.GetMessage(), "Hello "+clientMessage; got != want {
 		t.Errorf("r.GetMessage() = %v, want %v", got, want)
 	}
+	grpclog.Infof("client received message from server: %s", resp.GetMessage())
 
-	// Check the auth info.
-	authInfo, err := AuthInfoFromPeer(peer)
-	if err != nil {
-		t.Errorf("AuthInfoFromContext(peer) failed: %v", err)
-	}
-	s2aAuthInfo, ok := authInfo.(AuthInfo)
-	if !ok {
-		t.Errorf("authInfo is not an s2a.AuthInfo")
-	}
-	if got, want := s2aAuthInfo.AuthType(), authType; got != want {
-		t.Errorf("s2aAuthInfo.AuthType() = %v, want %v", got, want)
-	}
-	if got, want := s2aAuthInfo.ApplicationProtocol(), applicationProtocol; got != want {
-		t.Errorf("s2aAuthInfo.ApplicationProtocol() = %v, want %v", got, want)
-	}
-	if got, want := s2aAuthInfo.TLSVersion(), commonpb.TLSVersion_TLS1_3; got != want {
-		t.Errorf("s2aAuthInfo.TLSVersion() = %v, want %v", got, want)
-	}
-	if got, want := s2aAuthInfo.IsHandshakeResumed(), false; got != want {
-		t.Errorf("s2aAuthInfo.IsHandshakeResumed() = %v, want %v", got, want)
-	}
-	if got, want := s2aAuthInfo.SecurityLevel(), credentials.PrivacyAndIntegrity; got != want {
-		t.Errorf("s2aAuthInfo.SecurityLevel() = %v, want %v", got, want)
+	if EnableV2 == false {
+		// Check the auth info.
+		authInfo, err := AuthInfoFromPeer(peer)
+		if err != nil {
+			t.Errorf("AuthInfoFromContext(peer) failed: %v", err)
+		}
+		s2aAuthInfo, ok := authInfo.(AuthInfo)
+		if !ok {
+			t.Errorf("authInfo is not an s2a.AuthInfo")
+		}
+		if got, want := s2aAuthInfo.AuthType(), authType; got != want {
+			t.Errorf("s2aAuthInfo.AuthType() = %v, want %v", got, want)
+		}
+		if got, want := s2aAuthInfo.ApplicationProtocol(), applicationProtocol; got != want {
+			t.Errorf("s2aAuthInfo.ApplicationProtocol() = %v, want %v", got, want)
+		}
+		if got, want := s2aAuthInfo.TLSVersion(), commonpb.TLSVersion_TLS1_3; got != want {
+			t.Errorf("s2aAuthInfo.TLSVersion() = %v, want %v", got, want)
+		}
+		if got, want := s2aAuthInfo.IsHandshakeResumed(), false; got != want {
+			t.Errorf("s2aAuthInfo.IsHandshakeResumed() = %v, want %v", got, want)
+		}
+		if got, want := s2aAuthInfo.SecurityLevel(), credentials.PrivacyAndIntegrity; got != want {
+			t.Errorf("s2aAuthInfo.SecurityLevel() = %v, want %v", got, want)
+		}
 	}
 }
 
-func TestEndToEndUsingFakeS2AOverTCP(t *testing.T) {
```

Comment:
I created V1 and V2 versions of the all the TestXXX. I was thinking about converting each TestXXX into a for loop style test with two tests: EnableV2 == false and EnableV2 == true. Do you have any recommendation for this? Currently, there's a lot of code duplication when testing V1 and V2 in this file.

Reply:
In this case, let's keep them separate. I think there's some value in separating the v1 vs. v2 tests as much as possible. Thanks for checking!

Reply:
Sounds good, thanks for confirming!